src/bentoml/_internal/server/metrics/prometheus.py

```diff
@@ -108,59 +135,343 @@ def generate_latest(self):
             return self.prometheus_client.generate_latest()
 
     def text_string_to_metric_families(self) -> t.Generator[Metric, None, None]:
+        """
+        Parse Prometheus text format from a unicode string.
+
+        Returns:
+            Generator[Metric, None, None]: A generator of `Metric <https://prometheus.io/docs/concepts/metric_types/>`_ objects.
+        """
         yield from self.prometheus_client.parser.text_string_to_metric_families(
             self.generate_latest().decode("utf-8")
         )
 
     @property
     def CONTENT_TYPE_LATEST(self) -> str:
+        """
+        Returns:
+            str: Content type of the latest text format
+        """
         return self.prometheus_client.CONTENT_TYPE_LATEST
 
+    # For all of the documentation for instruments metrics below, we will extend
+    # upon prometheus_client's documentation, since their code segment aren't rst friendly, and
```

Comment:
I'm not really sure we should do this, to be honest, but I guess it is a little nicer.

Reply:
We probably want to commit upstream.

Since this is docs for our metrics, I can't find a nicer way to do this 😢